### PR TITLE
Pro 3550 declaration pdf

### PR DESCRIPTION
**Pull request by Wivenhoe-HMCTS**
### Change description ###
Addition of a hypertext link  on the declartion step to allow the user to view the legal declaration as a pdf document.

This PR is linked to a corresponding business-service change.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -136,12 +136,27 @@ router.get('/checkAnswersPdf', (req, res) => {
             res.setHeader('Content-disposition', 'attachment; filename=checkYourAnswers.pdf');
             res.send(result);
         })
+        .catch(err => {
+            res.status(500).render('errors/500', {cmmon: commonContent});
+        });
+    ;
+});
+
+router.get('/declarationPdf', (req, res) => {
```
Comment (shepda) [inline app/routes.js]: Would prefer this route to use lowercase with hyphens, for example `/declaration-pdf`

File: app/routes.js [hunk 2]
```diff
@@ -136,12 +136,27 @@ router.get('/checkAnswersPdf', (req, res) => {
...
+    const formdata = req.session.form;
+    services.createDeclarationPdf(formdata)
+        .then(result => {
+            res.setHeader('Content-Type', 'application/pdf');
+            res.setHeader('Content-disposition', 'attachment; filename=legalDeclaration.pdf');
```
Comment (shepda) [inline app/routes.js]: Would prefer this filename to use lowercase with hyphens, for example `legal-declaration.pdf`

File: app/steps/ui/declaration/template.html [hunk 3]
```diff
@@ -10,60 +10,65 @@ <h1 class="form-title heading-large">{{ content.title }}</h1>
...
             fields.declaration.value.understandItem2
         ]) }}
 
     </div>
-
+    <div class="declaration-grid-item">
+        <h2 class="heading-medium">{{ content.saveThisPage }}</h2>
+        <p><img class="icon-file-download" alt="Download PDF document"/>&nbsp;
+            <a id="declarationPdfHref" href="/declarationPdf">{{content.declarationPdf}}</a></p>
+    </div>
```
Comment (shepda) [inline app/steps/ui/declaration/template.html]: Shouldn't we be using a feature toggle for this link?

File: app/steps/ui/summary/template.html [hunk 4]
```diff
@@ -36,7 +36,7 @@ <h2 class="heading-medium">{{content.Summary.deceasedHeading}}</h2>
     </dl>
 
     <p><img class="icon-file-download" alt="Download PDF document"/>&nbsp;
-        <a id="checkAnswerHref" href="/checkAnswersPdf">{{content.Summary.checkAnswersPdf}}</a></p>
+    <a id="checkAnswerHref" href="/checkAnswersPdf">{{content.Summary.checkAnswersPdf}}</a></p>
```
Comment (shepda) [inline app/steps/ui/summary/template.html]: Shouldn't we be using a feature toggle for this link?

File: app/steps/ui/declaration/template.html [hunk 5]
```diff
@@ -63,8 +63,9 @@ <h2 class="heading-large declaration-subheader">{{ content.declarationHeader }}<
     </div>
     <div class="declaration-grid-item">
         <h2 class="heading-medium">{{ content.saveThisPage }}</h2>
-        <p><img class="icon-file-download" alt="Download PDF document"/>&nbsp;
-            <a id="declarationPdfHref" href="/declarationPdf">{{content.declarationPdf}}</a></p>
+        <p><img src="{{ asset_path }}images/icon-file-download.png?" alt="Download pdf" width="30" height="30"/>&nbsp;
```
Comment (shepda) [inline app/steps/ui/declaration/template.html]: We don't need to add this image manually as the GDS toolkit has a class that can add it so we should be using that instead

File: app/steps/ui/summary/template.html [hunk 6]
```diff
@@ -34,9 +34,11 @@ <h2 class="heading-medium">{{content.Summary.deceasedHeading}}</h2>
     <dl class="check-your-answers check-your-answers--long">
         {% include "ui/summary/includes/deceased.njk" %}
     </dl>
-
-    <p><img class="icon-file-download" alt="Download PDF document"/>&nbsp;
-    <a id="checkAnswerHref" href="/checkAnswersPdf">{{content.Summary.checkAnswersPdf}}</a></p>
+    <p>
+        <img src="{{ asset_path }}images/icon-file-download.png?" alt="Download pdf" width="30" height="30"/>
```
Comment (shepda) [inline app/steps/ui/summary/template.html]: We don't need to add this image manually as the GDS toolkit has a class that can add it so we should be using that instead

File: app/routes.js (hunk 1 above)
Comment (Wivenhoe-HMCTS) [reply app/routes.js]: ok, this is done and now resides within documentDownload.js

Comment (Wivenhoe-HMCTS) [reply app/routes.js]: ok, this has been done and now resides within documentDownload.js

File: app/routes.js (hunk 2 above)
Comment (Wivenhoe-HMCTS) [reply app/routes.js]: ok, done and now resides within documentDownload.js

File: app/steps/ui/summary/template.html (hunk 4 above)
Comment (Wivenhoe-HMCTS) [reply app/steps/ui/summary/template.html]: I refer to my earlier answer on deployment of the business service before the front end.

File: app/steps/ui/declaration/template.html (hunk 5 above)
Comment (Wivenhoe-HMCTS) [reply app/steps/ui/declaration/template.html]: yes, templtae has already been changed to use <i class="icon icon-file-download"></i>

File: app/steps/ui/summary/template.html (hunk 6 above)
Comment (Wivenhoe-HMCTS) [reply app/steps/ui/summary/template.html]: yes, template has already been changed to use

File: app/steps/ui/declaration/template.html (hunk 3 above)
Comment (Wivenhoe-HMCTS) [reply app/steps/ui/declaration/template.html]: It was discussed whether to use a feature but it was decided that we would deploy the corresponding business service pr prior too the front end

File: app/routes.js (hunk 2 above)
Comment (shepda) [reply app/routes.js]: Can this please be changed for `checkYourAnswers.pdf` and `legalDeclaration.pdf`

**Comment by Wivenhoe-HMCTS** [comment]
Closing due to jenkins pipeline issue

**Pull request closed without merging**